src/hostname/hostnamed.c

```diff
@@ -408,6 +409,62 @@ static int property_get_chassis(
 
         return sd_bus_message_append(reply, "s", name);
 }
+static const char * const dmi_invalid_list[] = {
+        "To be filled by OEM",
+        "TBD by OEM",
+        "Default String",
+        "Board Manufacturer",
+        "Board Vendor Name",
+        "Board Product Name",
+        NULL,	/* terminator */
+};
+
+static int is_dmi_valid(const char *field)
+{
+        int i = 0;
+
+        while (dmi_invalid_list[i]) {
+                if (strstr(field, dmi_invalid_list[i]))
+                        return 0;
+                i++;
+        }
+
+        return 1;
+}
```

Comment:
might as well change the return type to "bool" here, and return true/false.

should also filter out empty strings early on, maybe with an `if (isempty(field)) return false` line early on.